iot-hub/Samples/device/ConventionBasedOperations/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -110,12 +113,28 @@ public async Task PerformOperationsAsync(CancellationToken cancellationToken)
             }
         }
 
-        private async Task GetWritablePropertiesAndHandleChangesAsync()
+        private async Task GetWritablePropertiesAndHandleChangesAsync(CancellationToken cancellationToken)
         {
             ClientProperties properties = await _deviceClient.GetClientPropertiesAsync();
             ClientPropertyCollection writableProperties = properties.WritablePropertyRequests;
             long serverWritablePropertiesVersion = writableProperties.Version;
 
+            if (!writableProperties.Contains(Thermostat1))
+            {
+                // Update the reported property "targetTemperature" with the default values and ACK
+                // for the component "thermostat1" when its writable properties are empty.
+                // This is a component-level property update call.
+                await UpdateForEmptyWritableProperty(Thermostat1, cancellationToken);
+            }

```

Comment:
If I'm understanding this correctly, the intention here is to update the "reported" property value to some default value on initial start-up. If that is correct, then you don't need to call this from within the status change callback but rather only on the first initial client set up. If you call this from within the callback, then anytime the client reconnects the reported property value will be reset to the previously defined default value. Is that the intention here?

Reply:
@rido-min Do we want to report the default values for empty writable properties on **initial start-up only** or **each connection/reconnection** ?

Reply:
I'd say that defaultValues should be set only on *initial startup*, the problem is that during reconnection we don't know if the defaultValue has been successfully accepted. 

Ideally, we should get the twin on each connection/reconnection, compare the reported properties, and if they match the current values we should not write those again. 

To answer @abhipsaMisra question. The default values only make sense if the twin was empty (no previous reported/desired values found)

Reply:
In such a case I will move the report-default-values call to the client initialization instead.